job-runner/chain/traverser.go

```diff
@@ -295,22 +295,24 @@ func (t *traverser) Status() (proto.JobChainStatus, error) {
 // and attaches it to the job. When it's done, it sends the job out through the
 // doneJobChan.
 func (t *traverser) runJobs() {
-	for runJ := range t.runJobChan {
-		go func(j proto.Job) {
+	for runnableJob := range t.runJobChan {
+		go func(pJob proto.Job) {
+			// Always return the job when done, else the traverser will block.
+			defer func() { t.doneJobChan <- pJob }()
```

Comment:
is replacing the adding the job to the done job chain throughout the function with the defer here just a code clean up?

Reply:
Cleanup and improvement. Previously, iirc, there were returns on error before `pJob` was sent back, which means it never got sent back. But `pJob` should always be sent back regardless, so `defer` at the start is the solution.